lib/elixir/lib/task.ex

```diff
@@ -752,18 +789,29 @@ defmodule Task do
   a response waiting in the message queue this function will return
   `{:exit, :noproc}` as the result or exit reason can not be determined.
   """
-  @spec shutdown(t, timeout | :brutal_kill) :: {:ok, term} | {:exit, term} | nil
```

Comment:
The spec here was clear that `:infinity` was allowed.

Reply:
see the discussion above https://github.com/elixir-lang/elixir/pull/8139#discussion_r216138474

Reply:
On shutdown we send an exit signal. Passing `:infinity` means you are going to give it all the time it needs to shutdown. It doesn't mean we just wait, as a signal is sent anyway.

Some processes are guaranteed to shutdown, such as supervisors, so giving them time to terminate is fair.